activerecord/lib/active_record/errors.rb

```diff
@@ -178,18 +178,10 @@ class Rollback < ActiveRecordError
   class DangerousAttributeError < ActiveRecordError
   end
 
-  # Raised when unknown attributes are supplied via mass assignment.
-  class UnknownAttributeError < NoMethodError
-
-    attr_reader :record, :attribute
-
-    def initialize(record, attribute)
-      @record = record
-      @attribute = attribute.to_s
-      super("unknown attribute '#{attribute}' for #{@record.class}.")
-    end
-
-  end
+  UnknownAttributeError = ActiveSupport::Deprecation::DeprecatedConstantProxy.new( # :nodoc:
+    'ActiveRecord::UnknownAttributeError', 
+    'ActiveModel::AttributeAssignment::UnknownAttributeError'
```

Comment:
Leaving this undeprecated can cause confusion because while constant alias will work the caught exception instances will still have new name. People are just yelling about too long name space and want it to be sorter like in @egilburg's comment